test/extra/flask_test.py

```diff
@@ -0,0 +1,63 @@
+#-*- coding: utf-8 -*-
+#flask test
+#todo: It should be tested as django_test.py.
+import os, sys, subprocess, shutil
+import multiprocessing, time, signal
+import urllib2
+sys.path.append(os.path.dirname(__file__) + "/../lib")
+
+from test_helper import create_virtenv, run_test
+ENV_NAME = "flask_test_env_" + os.path.basename(sys.executable)
+SRC_DIR = os.path.abspath(os.path.join(ENV_NAME, "src"))
+PYTHON_EXE = os.path.abspath(os.path.join(ENV_NAME, "bin", "python"))
+
+def install_flask():
+    shutil.rmtree(SRC_DIR, ignore_errors=True)
+    os.makedirs(SRC_DIR)
+    subprocess.check_call(["git", "clone", "https://github.com/mitsuhiko/flask.git"], cwd=SRC_DIR)
```

Comment:
Some minor recommendations about this -- any reason we can't download these from PyPI?  `pip install flask` seemed to work -- the second argument to create_virtualenv() is a list of packages to install, so if you pass `["flask"]` it should do what you want.  If you really want to go to the git repo, you can do `["git+https://github.com/mitsuhiko/flask.git"]` and pip will do these steps for you.  But if there isn't any particular reason to pull them from git, I'd recommend switching this to PyPI since that should be faster.  If you do that, `install_flask()` can just become `sys.path.append(os.path.join(ENV_NAME, "site-packages"))`.

One additional consideration (regardless of the git vs pypi decision) is that we want to pin the version of every library we test.  It's quite annoying when someone updates their library in a way that breaks our test, and our test picks it up since it is configured to pull whatever the latest version is.  If you switch this to pip, please try running `pip install flask werkzeug jinja2...` to see what versions it installs, and then pin the dependencies like `create_virtualenv(ENV_NAME, ["flask==0.0.0", "werkzeug==9.9.9", ...])`.